tensorflow_model_optimization/g3doc/guide/clustering/clustering_comprehensive_guide.ipynb

```diff
@@ -0,0 +1,433 @@
+{
```

Comment:
<p>If pre-trained weights are not supported, I don't see how a user can use this with clustering, which requires pre-trained weights.</p><p> </p><p>That said, see the subclassed model example from <a href="https://github.com/tensorflow/model-optimization/pull/272" target="_blank">https://github.com/tensorflow/model-optimization/pull/272</a> and the testing from <a href="https://github.com/tensorflow/model-optimization/pull/383" target="_blank">https://github.com/tensorflow/model-optimization/pull/383</a>. That should work for you also.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/model-optimization/pull/330/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
<blockquote>If pre-trained weights are not supported, I don't see how a user can use this with clustering, which requires pre-trained weights.</blockquote><p>That was copied blindly from elsewhere. Sorry.</p><blockquote>That said, see the subclassed model example&nbsp;</blockquote><p>OK, added it.</p><p>There is a question on this: in the example, if MyModel had been inherited from another 'Model' subclass that had clusterable layers, those layers wouldn't have been clustered, right? Do we need to mention this in the guide? I do not see it in the pruning or QAT example, so I assume no? Then I am not sure how pruning or QAT handles this.</p>

Reply:
<p>So the statement for clustering, subclassed models are not supported should porbably be more like they are not&nbsp;<em>fully</em>&nbsp;supported?</p>